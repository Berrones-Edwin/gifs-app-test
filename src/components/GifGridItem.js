import Proptypes from 'prop-types'
import React from 'react'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p> {title} </p>
    </div>
  )
}

GifGridItem.prototypes = {
  title: Proptypes.string.isRequired,
  url: Proptypes.string.isRequired
}
