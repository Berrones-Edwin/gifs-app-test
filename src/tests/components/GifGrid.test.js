import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import React from 'react'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('test in component <GifGrid />', () => {
  test('should show component successfully', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })
    const wrapper = shallow(<GifGrid category={'hulk'} />)

    expect(wrapper).toMatchSnapshot()
  })
  test('should show images when use custom hook useFetchGifs', () => {

    const mockData = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa',
        title: 'my title'
      }
    ]
    useFetchGifs.mockReturnValue({
      data: mockData,
      loading: false
    })
    const wrapper = shallow(<GifGrid category={'hulk'} />)

    // expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(mockData.length)
  })
})
