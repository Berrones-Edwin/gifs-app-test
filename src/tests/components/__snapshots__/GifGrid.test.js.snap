// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in component <GifGrid /> should show component successfully 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    hulk
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
